src/templates/ComponentTemplate.jsx

```diff
@@ -1,136 +1,199 @@
 import { Accordion } from "@trussworks/react-uswds";
 import PropTypes from "prop-types";
 import { Link } from "react-router-dom";
-import { isEmpty } from "../utils";
 import ResponsibilityBox from "../atoms/ResponsibilityBox";
+import { CATALOG_DISPLAY_NAMES } from "../constants";
 import ComponentProjectForm from "../organisms/ComponentProjectForm";
 import ControlHeader from "../organisms/ControlHeader";
 
 export function ComponentTemplate({
-  component,
-  selectedControl,
   catalogData,
+  catalogVersion,
+  componentData,
+  componentId,
+  projectData,
+  selectedControl,
+  handleSelectControl,
   handleProjectUpdate,
 }) {
-  const { title, description, standard, source } = component.component_data;
+  const { title, description } = componentData;
+  const source = componentData.standards[catalogVersion].source;
+  const componentControls = componentData.standards[catalogVersion].controls;
 
-  const accordionItemsProps = [
-    {
-      title: "CMS Implementation Standards",
-      content: (
-        <p>
-          {selectedControl.implementationStandards ||
-            "No implementation standards found for this control."}
-        </p>
-      ),
-      expanded: false,
-      id: "implementation_standards",
-      headingLevel: "h3",
-    },
-    {
-      title: "CMS Control Guidance",
-      content: (
-        <p>
-          {selectedControl.guidance ||
-            "No control guidance found for this control."}
-        </p>
-      ),
-      expanded: false,
-      id: "control_guidance",
-      headingLevel: "h3",
-    },
-    {
-      title: "Narrative",
-      content: <p>{selectedControl.narrative}</p>,
-      expanded: true,
-      id: "inherited_narratives",
-      headingLevel: "h3",
-    },
-  ];
+  const selectedControlCatalogData = {
+    low: catalogData[catalogVersion].low.controls[selectedControl],
+    moderate: catalogData[catalogVersion].moderate.controls[selectedControl],
+    high: catalogData[catalogVersion].high.controls[selectedControl],
+  };
+  const selectedControlComponentData = componentControls[selectedControl];
+
+  const getImplementationText = (level) => {
+    if (
+      selectedControlCatalogData &&
+      selectedControlCatalogData[level].implementationStandards
+    ) {
+      return selectedControlCatalogData[level].implementationStandards;
+    } else {
+      return "No implementation standards found for this control.";
+    }
+  };
+
+  const getControlGuidanceText = (level) => {
+    if (
```

Comment:
Praise: I really appreciate (from you) adding these guard clauses since we're constantly learning that there is so much variation in the catalog data which I don't appreciate.